larray_editor/arraymodel.py

```diff
@@ -299,7 +299,14 @@ def set_format(self, format):
         self.reset()
 
     def set_background(self, bg_gradient=None, bg_value=None):
-        self.bg_gradient = bg_gradient
+        if bg_gradient is not None and not isinstance(bg_gradient, LinearGradient):
+            raise ValueError("Expected None or LinearGradient instance for `bg_gradient` argument")
+        if bg_value is not None and not (isinstance(bg_value, np.ndarray) and bg_value.ndim == 2):
+            raise ValueError("Expected None or 2D Numpy ndarray with shape {} for `bg_value` argument"
+                             .format(self._data.shape))
```

Comment:
you use the precise shape in the error message but not in the check. I guess you should replace bg_value.ndim == 2 by bg_value.shape == self._data.shape.

Reply:
Yes, I should.